[clang-tidy] fix false negative in cppcoreguidelines-missing-std-forward (#83987)
llvm/llvm-project: clang-tools-extra/docs/ReleaseNotes.rst

- Improved :doc:`cppcoreguidelines-missing-std-forward
  <clang-tidy/checks/cppcoreguidelines/missing-std-forward>` check by no longer

>   giving false positives for deleted functions and fix false negative when one
>   parameter is forwarded, but some other parameter isn't.

Reviewer: ```suggestion
  giving false positives for deleted functions and fix false negative when some
  parameters are forwarded, but other aren't.
```

Author: Thanks for review to make the description more clear!
I think you can check https://github.com/llvm/llvm-project/blob/2a1b09fee4b4b22f4f7189695ce3e858b91a8d69/clang-tools-extra/test/clang-tidy/checkers/cppcoreguidelines/missing-std-forward.cpp#L171-L173
this testcase may helpful for solving confusions.
